### ISSUE TITLE
get notes from a sighted player on what things might be problematic for space age support

### ISSUE THREAD

**Issue by ahicks92** [Factorio-Access/FactorioAccess#276]
We need someone with mod familiarity to complete space age and come back and infodump a bunch of info on what might be problematic and where.  Some issues are already being filed but we need the full set, and in particular I will call out space platforms.

**Comment by EphDoering** [comment]
Not specific to space age, but space age makes the problem much worse on at least 3 of the planets. Finding a nice area to build in is difficult. Cliff explosives get pushed back till metallurgic science, and Vulcanis cliffs in the cliff biome makes it really hard to find suitable build spaces. The lava biome is similarly terrible. Fulgora has small islands so trying to figure out what will fit where would be a huge pain. Aquillo might not actually be as bad as I was originally, thinking since you can build your own ice, but you'll need some minimal production before you can get started with the ice platform production I think.

In any case, I don't have any proposals yet to help with this issue, I'm just raising that I know finding buildable places was already hard in 1.1 and it will be much more challenging in Space Age.

**Comment by ahicks92** [comment]
Well, I've been thinking about this wrt fulgora specifically.  When I wrote the incremental floodfill algorithm, I had that in mind.  We can invert the conditions and get approximate bounding boxes of islands no trouble--it's even fast.  We likely can detect the inversion requirement even, e.g. just check "is map more than 50% water" or whatever (which gets seablock).

New scanner also made large cursor sizes not lag, so you can zoom out and move around up to 251x251, but from past experience this isn't enough for cliffs.

In #218 (probably long forgotten by anyone but me) I presented the idea of a map searcher: you add a list of conditions and it finds spots (not exhaustively, just some, for performance).  Think train schedules but the conditions are "iron ore within 50 tiles, not cliffs within 50 tiles" style conjunctions.  Perhaps we should think about reprioritizing it but it's one of the things that's really hard to do without good ways to write UI stuff, in the sense that we could but it'd be kudzu all over the codebase to get all the keyboard hooks etc.

I don't know how to do it efficiently at the moment for a variety of reasons but we can likely also tag tiles with the bounding boxes built such that the bounding box contains zero cliffs or whatever.  I'm not feeling clever enough at the moment to think of an efficient algorithm for it save perhaps binary searching on the box size: the issue is that we can't hold a box per tile, and recomputing that wouldn't be cheap to say the least.

In other words I've been giving that general problem some thought, though as regards space age specifically I personally have no problem turning off cliffs and saying sorry we're working on it while we go do more important things.

**Comment by ahicks92** [comment]
Closing as fixed based on analysis.
